opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -351,20 +353,15 @@ where
         .iter()
         .map(|log_data| (&log_data.0, &log_data.1))
         .collect();
-    let export = exporter.export(LogBatch::new(log_vec.as_slice()));
-    let timeout = runtime.delay(time_out);
-    pin_mut!(export);
-    pin_mut!(timeout);
-    match future::select(export, timeout).await {
-        Either::Left((export_res, _)) => export_res,
-        Either::Right((_, _)) => {
-            otel_error!(
-                name: "export_with_timeout_timeout",
-                timeout_duration = time_out.as_millis()
-            );
-            ExportResult::Err(LogError::ExportTimedOut(time_out))
-        }
-    }
+    let _export = exporter.export(LogBatch::new(log_vec.as_slice()));
+    // let timeout = runtime.delay(time_out);
+    pin_mut!(_export);
+    // pin_mut!(timeout);
+    // match future::select(export, export).await {
+    //     Either::Left((export_res, _)) => export_res,
+    //     Either::Right((_, _)) => ExportResult::Err(LogError::ExportTimedOut(time_out)),
+    // }
+    ExportResult::Ok(())
```

Comment:
So, we are expecting export method to complete in a definite time-period, and we are not handling timeout outside of export method ? any reason why it is so ?

Reply:
yes. Just chatted with @ThomsonTan about this. If a rogue exporter takes forever, then BatchProcessor's will get stuck behind that. For now, lets us document this limitation and move on.
(Metrics have the same problem. Its slightly worse in metrics, as observable callbacks can also go rogue and entire Metrics SDK will halt to function)